wal/src/message_queue_impl/region_meta.rs

```diff
@@ -339,6 +345,118 @@ impl OffsetRange {
     }
 }
 
+/// Builder for `RegionMeta`
+#[allow(unused)]
+#[derive(Debug, Default)]
+pub struct RegionMetaBuilder {
+    table_metas: HashMap<TableId, TableMetaInner>,
+}
+
+#[allow(unused)]
+impl RegionMetaBuilder {
+    pub fn apply_region_meta_snapshot(&mut self, snapshot: RegionMetaSnapshot) -> Result<()> {
+        debug!("Apply region meta snapshot, snapshot:{:?}", snapshot);
+
+        for entry in snapshot.entries.into_iter() {
+            ensure!(self.table_metas.insert(entry.table_id, entry.clone().into()).is_none(),
+                Build { msg: format!("apply snapshot failed, shouldn't exist duplicated entry in snapshot, duplicated entry:{:?}", entry) }
+            );
+        }
+
+        Ok(())
+    }
+
+    pub fn apply_region_meta_delta(&mut self, delta: RegionMetaDelta) -> Result<()> {
+        debug!("Apply region meta delta, delta:{:?}", delta);
+
+        let mut table_meta_inner = self
```

Comment:
Shall we just call this variable as `table_meta`? It is weird to suffix it with `inner`.